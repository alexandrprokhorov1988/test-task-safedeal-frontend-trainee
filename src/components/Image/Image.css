.photo-grid-img {
  min-width: 229px;
  width: 100%;
  min-height: 142px;
  object-fit: cover;
  cursor: pointer;
}

.photo-grid-img:hover {
  opacity: 0.9;
}
