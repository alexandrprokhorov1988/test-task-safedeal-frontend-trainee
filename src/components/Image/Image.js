import React from 'react';
import './Image.css';
import PropTypes from "prop-types";

const Image = React.memo(({ id, url, onImageClick }) => {

  function handleImageClick() {
    onImageClick(id);
  }

  function handleKeyDown(e) {
    if (e.key === 'Enter') {
      onImageClick(id);
    }
    return false;
  }

  return (
    <img
      key={id}
      src={url}
      alt="Картинка"
      className="photo-grid-img"
      onClick={handleImageClick}
      onKeyDown={handleKeyDown}
      tabIndex={0}
      role="presentation"
    />
  );
});

Image.propTypes = {
  id: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
  onImageClick: PropTypes.func.isRequired
};

Image.defaultProps = {
  onClick: () => {
  },
  onKeyDown: () => {
  },
  id: "",
  url: ""
};

export default Image;
