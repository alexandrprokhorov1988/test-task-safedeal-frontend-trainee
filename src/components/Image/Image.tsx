import React from 'react';
import './Image.css';

interface IImage {
  id: number;
  url: string;
  onImageClick: (id: number) => void;
}

const Image: React.FC<IImage> = ({id, url, onImageClick}) => {

  function handleImageClick(): void {
    onImageClick(id);
  }

  function handleKeyDown(e: React.KeyboardEvent): void | boolean {
    if (e.key === 'Enter') {
      onImageClick(id);
    }
    return false;
  }

  return (
    <img
      key={id}
      src={url}
      alt="Картинка"
      className="photo-grid-img"
      onClick={handleImageClick}
      onKeyDown={handleKeyDown}
      tabIndex={0}
      role="presentation"
    />
  );
};

export default React.memo(Image);
