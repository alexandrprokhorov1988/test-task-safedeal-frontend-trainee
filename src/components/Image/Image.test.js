import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import {act} from "react-dom/test-utils";
import Image from './Image';
import handleGetOriginalSizeImage from '../../components/App/App';

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Image component', () => {
  it('Должна вызывать коллбэк функцию по click', () => {
    const onChange = jest.fn();
    act(() => {
      render(<Image id={237} url={'#'} onImageClick={onChange}/>, container);
    });
    const image = document.querySelector(".photo-grid-img");
    act(() => {
      image.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('Должна вызывать коллбэк функцию по enter', () => {
    const onChange = jest.fn();
    act(() => {
      render(<Image id={237} url={'#'} onImageClick={onChange}/>, container);
    });
    const image = document.querySelector(".photo-grid-img");
    act(() => {
      image.dispatchEvent(new KeyboardEvent("keydown", { bubbles: true, key: 'Enter' }));
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('Image snapshot', () => {
    const render = renderer
      .create(<Image id={237} url={'#'} onImageClick={handleGetOriginalSizeImage}/>)
      .toJSON();
    expect(render).toMatchSnapshot();
  });
});
