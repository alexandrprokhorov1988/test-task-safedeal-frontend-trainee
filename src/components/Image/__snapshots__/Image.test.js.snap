// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Image snapshot 1`] = `
<img
  alt="Картинка"
  className="photo-grid-img"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  src="#"
  tabIndex={0}
/>
`;

exports[`Image snapshot 1`] = `
<img
  alt="Картинка"
  className="photo-grid-img"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  src="#"
  tabIndex={0}
/>
`;

exports[`Рендер footer с текстом jestio 1`] = `
<img
  alt="Картинка"
  className="photo-grid-img"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  src="#"
  tabIndex={0}
/>
`;
